haskell_scotty/Server.hs

```diff
@@ -1,24 +1,15 @@
 {-# LANGUAGE OverloadedStrings #-}
-import Web.Scotty
-import Network.HTTP.Types.Status
-
--- | Build a basic test page.
-helloTestPage :: ActionM ()
-helloTestPage = html "Hello world"
-
--- | Build a test page and do some status stuff.
-compoundTestPage :: ActionM ()
-compoundTestPage = do
-  html "You are not supposed to be here."
-  status badRequest400
 
--- | Specify the behavior of each API endpoint.
-router :: ScottyM ()
-router = do
-  get "/api/test" helloTestPage
-  get "/api/test2" compoundTestPage
+import Web.Scotty
+import FetchClients
+import Control.Monad.IO.Class
 
--- | Construct the actual webserver.
 main :: IO ()
-main = scotty 6969 router
+main = scotty 6969 $ do
+  get "/api/test" $ do
+    html "Hello world"
+
+  get "/api/clients" $ do
+    clients <- liftIO fetchClients
```

Comment:
Ah, that makes sense, thank you!

The reason I keep changing the top-level to the Sinatra-style syntax is to ease comparison with impls in other Sinatra clones, as we have several in this repo. I can see how this clashes with a general desire to be explicit in Haskell for clarity.